pkg/dynamicinstrumentation/eventparser/event_parser.go

```diff
@@ -224,18 +238,38 @@ func countBufferUsedByTypeDefinition(root *ditypes.Param) int {
 		front := queue[0]
 		queue = queue[1:]
 		counter += 3
-		queue = append(queue, front.Fields...)
+
+		if reflect.Kind(front.Kind) == reflect.Slice && len(front.Fields) > 0 {
+			// The fields of slice elements are amended after the fact to account
+			// for the runtime discovered length. However, only one definition of
+			// the slice element's type is present in the buffer.
+			queue = append(queue, front.Fields[0])
+		} else {
+			queue = append(queue, front.Fields...)
+		}
 	}
 	return counter
 }
 
 func isTypeWithHeader(pieceType byte) bool {
 	return reflect.Kind(pieceType) == reflect.Struct ||
-		reflect.Kind(pieceType) == reflect.Slice ||
 		reflect.Kind(pieceType) == reflect.Array ||
+		reflect.Kind(pieceType) == reflect.Slice ||
 		reflect.Kind(pieceType) == reflect.Pointer
 }
 
+func isRuntimeSizedType(pieceType byte) bool {
+	return reflect.Kind(pieceType) == reflect.Slice ||
+		reflect.Kind(pieceType) == reflect.String
+}
+
+func readRuntimeSizedLength(lengthBytes []byte) (uint16, error) {
+	if len(lengthBytes) != 2 {
+		return 0, errors.New("malformed bytes for runtime sized length")
+	}
+	return binary.NativeEndian.Uint16(lengthBytes), nil
```

Comment:
Is `NativeEndian` intentional here? We use `LittleEndian` everywhere else

Reply:
Yes, NativeEndian resolves to LittleEndian on little endian machines. I've made sure this is consistent now.